mypy/fastparse.py

```diff
@@ -87,6 +90,35 @@ def parse_type_comment(type_comment: str, line: int) -> Type:
         return TypeConverter(line=line).visit(typ.body)
 
 
+def parse_docstring(docstring: str, arg_names: List[str],
+                    line: int) -> Optional[Tuple[List[Type], Type]]:
+    """Parse a docstring and return type representations.
+
+    Returns a 2-tuple: (list of arguments Types, and return Type).
+    """
+    def pop_and_convert(name):
+        t = type_map.pop(name, None)
+        if t is None:
+            return AnyType()
+        elif isinstance(t, Type):
+            return t
+        else:
+            return parse_str_as_type(t, line)
+
+    docstring_parser = hooks.get_docstring_parser()
+    if docstring_parser is not None:
+        type_map = docstring_parser(inspect.cleandoc(docstring), line)
```

Comment:
I still believe calling cleandoc() is up to the custom parser. They may have some other approach to parsing the contents of the docstring.